pyro/infer/mcmc/nuts.py

```diff
@@ -51,6 +55,10 @@ class NUTS(HMC):
     :param bool adapt_mass_matrix: A flag to decide if we want to adapt mass
         matrix during warm-up phase using Welford scheme.
     :param bool full_mass: A flag to decide if mass matrix is dense or diagonal.
+    :param bool use_multinomial_sampling: A flag to decide if we want to sample
+        candidates along its trajectory using "multinomial sampling" or using
+        "slice sampling". Slice sampling is used in the original NUTS paper [1],
+        while multinomial sampling is suggested in [2].
```

Comment:
We should mention that this is the default, and if set to False, the sampler uses slice sampling instead.